### PR TITLE
Add taglist preview mode

### PR DESCRIPTION
**Pull request by Rike-cz**
#4330 solution.

### CODE REVIEW

**Comment by bennothommo** [comment]
@Rike-cz I've tested this and it works well. The one thing I would suggest is that the LESS code be moved to an asset file within the widget files, or to the backend LESS files somewhere. Where it is currently (within the Select2 styling) isn't very obvious.

**Comment by w20k** [comment]
@bennothommo why? Those styles are related to OCMS select2 improvements/customizations within the core. I don't see any point of moving those. 
I've also tested it. Works great, thanks @Rike-cz!

**Comment by w20k** [comment]
The only thing I'd really want to change, based on the BEM methodology. @Rike-cz could add something like `--preview` postfix`.tag--preview` to differentiate the state.

**Comment by bennothommo** [comment]
@w20k Put it this way - if we upgrade Select2 one day and have to update the CSS/LESS along with it, are we going to remember that the taglist styling is at the bottom of that file?

**Comment by w20k** [comment]
@bennothommo we need to update, if nothing in structure will changes, this file '@import "../vendor/select2/css/select2.less";'

**Comment by Rike-cz** [comment]
I think taglist is derived from Select2, so that's why we have stylesheets in one file.

File: modules/system/assets/ui/less/select.less [hunk 1]
```diff
@@ -588,4 +588,25 @@
 
 .form-inline .select2-container--default {
     display: inline-block;
+}
+
+// Taglist preview mode
+//------------------------------------
+
+.taglist {
```
Comment (w20k) [inline modules/system/assets/ui/less/select.less]: @Rike-cz could you add `--preview` posfix? Thanks!

**Comment by bennothommo** [comment]
@w20k Sorry, I didn't see that it links to a vendor LESS file, I presumed `modules/system/assets/ui/less/select.less` had all the Select2 code in it. I still think it might make sense in its own LESS file just purely for separation of concerns (in preview mode, the taglist isn't a Select2), but I'm not going to stand in the way if you guys disagree with me :)

**Comment by LukeTowers** [comment]
I don't think it would be a bad idea to have the taglist specific CSS in its own file under its own directory. Most of the other formwidgets operate that way so I think it would be fine to do it that way for this too. In this case it's a small enough addition that it's not a huge deal either way, but I would say that the "proper" way of doing it is to have the taglist specific stuff in its own file under the taglist formwidget folder structure.

**Comment by bennothommo** [comment]
@Rike-cz Last step, could you run `artisan october:util compile less` and provide the compiled `storm.css` file as part of your PR? Should be good to merge once done. :)

**Comment by w20k** [comment]
Just minor note, why I don't like the idea of separation, either we should move everything from `select2.less` or leave as it is. Because right now it's just an override to `select2.less` styles.

Few examples:
_Dorpdown_
https://github.com/octobercms/october/blob/dd964fe3991abebfadd6c280dd24f98089da2201/modules/system/assets/ui/less/select.less#L218
_Single Select_
https://github.com/octobercms/october/blob/dd964fe3991abebfadd6c280dd24f98089da2201/modules/system/assets/ui/less/select.less#L251
_Multi Select_
https://github.com/octobercms/october/blob/dd964fe3991abebfadd6c280dd24f98089da2201/modules/system/assets/ui/less/select.less#L251
And so on.

**Comment by bennothommo** [comment]
@w20k I think those ones are fine because, as you said, they are overrides to the original Select2 styling to theme them for October. The taglist preview however is *not* a Select2 dropdown, it's simply styled in a way that it looks like a Select2. :)

**Comment by w20k** [comment]
Sorry, @Rike-cz did my best, but @bennothommo is right on this one :) My bad, haven't reviewed `Php` part.

**Pull request merged**